sparse/impl/KokkosSparse_par_ilut_numeric_impl.hpp

```diff
@@ -440,17 +461,25 @@ struct IlutWrap {
    * make this function determistic, but that could cause par_ilut
    * to take longer (more iterations) to converge.
    */
-  template <class ARowMapType, class AEntriesType, class AValuesType,
-            class LRowMapType, class LEntriesType, class LValuesType,
-            class URowMapType, class UEntriesType, class UValuesType,
-            class UtRowMapType, class UtEntriesType, class UtValuesType>
-  static void compute_l_u_factors(
+  template <bool async_update, class ARowMapType, class AEntriesType,
+            class AValuesType, class LRowMapType, class LEntriesType,
+            class LValuesType, class URowMapType, class UEntriesType,
+            class UValuesType, class UtRowMapType, class UtEntriesType,
+            class UtValuesType>
+  static void compute_l_u_factors_impl(
       IlutHandle& ih, const ARowMapType& A_row_map,
       const AEntriesType& A_entries, const AValuesType& A_values,
       LRowMapType& L_row_map, LEntriesType& L_entries, LValuesType& L_values,
       URowMapType& U_row_map, UEntriesType& U_entries, UValuesType& U_values,
       UtRowMapType& Ut_row_map, UtEntriesType& Ut_entries,
-      UtValuesType& Ut_values, const bool async_update) {
+      UtValuesType& Ut_values_arg) {
+    // UtValues needs to be Atomic if async updates are on. Otherwise,
+    // non-atomic is fine.
+    using UtValuesSafeType = typename ParIlutOnly::UtViewType<
+        async_update>::template UtValuesViewType<UtValuesType>;
```

Comment:
```suggestion
    using UtValuesSafeType = std::conditional_t<async_update,
      Kokkos::View<typename UtValuesType::non_const_value_type*, typename UtValuesType::array_layout, typename UtValuesType::device_type, Kokkos::MemoryTraits<Kokkos::Unmanaged | Kokkos::RandomAccess | Kokkos::Atomic> >,
      UtValuesType>;
```
std::conditional is like the "a ? b : c" operator but for types, so it's equivalent to what you had but lets you skip writing out the true/false specializations.

Reply:
Very cool! Thanks